analyses/npl/03_flood_events.md

```diff
@@ -0,0 +1,327 @@
+SHAPEFILE_DIR = DATA_DIR_PUBLIC / 'raw/npl/cod_ab'
+
+PAST_EVENTS_FILENAME = RCO_DIR / 'NepalHistoricalFlood1971-2020.xlsx' 
+BASINS_SHAPEFILE = RCO_DIR / 'shapefiles/Major_River_Basins.shp'
+WATERSHED_SHAPEFILE = RCO_DIR / 'shapefiles/Major_watershed.shp'
+ADMIN_SHAPEFILE = SHAPEFILE_DIR / 'npl_admbnda_ocha_20201117/npl_admbnda_nd_20201117_shp.zip'
+ADMIN2_SHAPEFILE = 'npl_admbnda_districts_nd_20201117.shp'
+
+EVENTS_FOR_RAGINDRA = DATA_DIR_PUBLIC / 'exploration/npl/glofas/npl_glofas_events.xlsx'
+```
+
+### Read in data and clean slightly
+
+```python
+ds_glofas_reanalysis = utils.get_glofas_reanalysis(
+    country_iso3=COUNTRY_ISO3)
+df_return_period = utils.get_return_periods(ds_glofas_reanalysis)
+```
+
+```python
+# Read in events and admin, 
+# make pcode column names match for simplicity
+df_events = (pd.read_excel(PAST_EVENTS_FILENAME)
+             .rename(columns={'DIST_CODE_ETHOS': 'pcode'}))
+df_admin = (gpd.read_file(f'zip://{ADMIN_SHAPEFILE}!{ADMIN2_SHAPEFILE}')
+            .rename(columns={'DIST_PCODE': 'pcode'}))
+```
+
+```python
+# Cleaning events data
+# Drop events with no date and convert date column
+df_events = df_events.loc[df_events['Year'] > 0]
+# For days that are 0, assume they mean the middle of the month
+df_events['Day'] = np.where(df_events['Day'] == 0, 15, df_events['Day'])
+df_events['Incident Date'] = pd.to_datetime(df_events[['Year', 'Month', 'Day']])
+
+# Only select events in the time range of GloFAS
+df_events = df_events.loc[df_events['Incident Date'] > ds_glofas_reanalysis.time[0].data]
+```
+
+```python
+# Plot the number of events per destrict
+x = df_events.rename(columns={'Year': 'num_events'}).groupby('pcode').count()['num_events']
+df_admin.join(x, on='pcode', how='left').plot(column='num_events', legend=True)
+```
+
+```python
+# Read in basin files
+df_basins = gpd.read_file(BASINS_SHAPEFILE)
+df_watershed = gpd.read_file(WATERSHED_SHAPEFILE)
+```
+
+### Get sub-set of events based on basin pcodes
+
+```python
+# Create the target region file 
+df_target_regions = gpd.GeoDataFrame(columns=["name", "districts", "geometry"]).set_crs('EPSG:4326')
+for basin in ["Karnali", "Babai", "Bagmati", "West Rapti"]:
+    df_target_regions = df_target_regions.append(
+    {"name": basin, 
+    "geometry": df_basins.loc[df_basins['Major_Basi'] == f'{basin} River Basin', 'geometry'].iloc[0]},
+    ignore_index=True
+    )
+# Get intersection between Saptakoshi Watershed and Koshi Basin
+saptakoshi = df_watershed.loc[df_watershed['WSH_NME'] == 'Saptakoshi Watershed', 'geometry'].iloc[0]
+koshi = df_basins.loc[df_basins['Major_Basi'] == 'Koshi River Basin', 'geometry'].iloc[0]
+
+df_target_regions = df_target_regions.append(
+    {"name": "Koshi", 
+    "geometry": koshi.union(saptakoshi)},
+    ignore_index=True
+)
+```
+
+```python
+# For each basin, get a list of affected regions
+# Must intersect 25%
+intersection_thresh = 0.25
+df_target_regions["districts"] = df_target_regions['geometry'].apply(lambda x:
```

Comment:
Might be worth documenting the justification for 25% overlapping area here.

Reply:
I added a comment -- it's not overly rigorous, but this analysis is not used for anything anyway